Add groups as an omittable field

Elchi3 opened the pull request:
Illustration for https://github.com/web-platform-dx/web-features/issues/648

foolip reviewed (commented):
I'd like to treats groups like specs and make it a string or list or strings depending on how many there are. I think a single group is going to be common, probably more common than multiple groups. So I'd call it `group` in singular just like `spec`.

Elchi3 commented:
Updated to be consistent with the `spec` field. Like for `spec`, when there is just one entry, I've removed the yaml list. So, If this sees adaption in the schema, we will need something like this:

```
"group": {
  "anyOf": [
    {
      "description": "Group identifier",
      "type": "string"
    },
    {
      "items": {
        "description": "Group identifier",
        "type": "string"
      },
      "minItems": 2,
      "type": "array"
    }
  ],
  "description": "Group"
}, 
```

foolip commented:
Since it's not published and doesn't get in the way, I'd be happy to merge this. We're still discussing how to represent group properties (name/description/etc.) and group hierarchy in https://github.com/web-platform-dx/web-features/issues/648, but we can do that in a separate PR.
